Added google firebase authentication

IsaackMwenda opened the pull request:
Created a Login page
Added google firebase authentication
![image](https://user-images.githubusercontent.com/23135322/49514130-37d79380-f8a4-11e8-8c43-baafcd3de5fd.png)

IsaackMwenda committed:
Added google firebase authentication

Created a Login page
Added google firebase authentication

lukechurch commented:
I think you'll also need to adjust rules.json at the same time as this? You can do that by changing it in this PR

IsaackMwenda committed:
Added firestore rules,fixed page flickering

Added firestore.rules
Fixed page flickering on data update
Fixed identation and whitespaces
Added a firebase.js file containing firebase API statements
Added a favicon

IsaackMwenda commented:
I have Added firestore.rules
Fixed page flickering on data update
Fixed identation and whitespaces
Added a firebase.js file containing firebase API statements
Added a favicon

IsaackMwenda committed:
'Added columns for wrong scheme and not codable messages'

IsaackMwenda commented:
Added columns for wrong scheme and not codable messages i.e (Number of messages and %)
![image](https://user-images.githubusercontent.com/23135322/49650417-25dd2880-fa3d-11e8-950d-83da57bc5478.png)

lukechurch commented:
> Added columns for wrong scheme and not codable messages i.e (Number of messages and %)
> ![image](https://user-images.githubusercontent.com/23135322/49650417-25dd2880-fa3d-11e8-950d-83da57bc5478.png)

Thanks very much for this @IsaackMwenda, impressive! - however this is conceptually separate from authentication so in future on a separate branch and PR please. (cc @ritazagoni )

lukechurch committed:
Update coding_progress.html

lukechurch commented on public/coda/firestore.rules:
```diff
@@ -0,0 +1,16 @@
+      }
+    }
+  }
+  
```
Sorry @IsaackMwenda I don't think this works, I'm going to remove it from the PR

lukechurch committed:
Delete firestore.rules
